packages/cli/src/checks/__tests__/EXTERNAL_MISMATCH.ts

```diff
@@ -224,9 +224,82 @@ it("should not error if the value is not a valid semver range", () => {
   };
   ws.set("pkg-2", pkg2);
 
-  let errors = internalMismatch.validate(pkg2, ws, rootWorkspace, {});
+  let errors = externalMismatch.validate(pkg2, ws, rootWorkspace, {});
   expect(errors.length).toEqual(0);
 
-  errors = internalMismatch.validate(ws.get("pkg-1")!, ws, rootWorkspace, {});
+  errors = externalMismatch.validate(ws.get("pkg-1")!, ws, rootWorkspace, {});
   expect(errors.length).toEqual(0);
 });
+
+it("should not error if the range is included in the allowedDependencyVersions option", () => {
+  let ws = getWS();
+
+  ws.get("pkg-1")!.packageJson.dependencies = { something: "1.0.0" };
+
+  let pkg2 = getFakeWS("pkg-2");
+  pkg2.packageJson.dependencies = {
+    something: "2.0.0"
+  };
+  ws.set("pkg-2", pkg2);
+
+  const options = {
+    allowedDependencyVersions: {
+      something: ["1.0.0", "2.0.0"]
+    }
+  };
+
+  let errors = externalMismatch.validate(pkg2, ws, rootWorkspace, options);
+  expect(errors.length).toEqual(0);
+
+  errors = externalMismatch.validate(
+    ws.get("pkg-1")!,
+    ws,
+    rootWorkspace,
+    options
+  );
+  expect(errors.length).toEqual(0);
+});
+
+it("should error if the range is the range is outside allowedDependencyVersions and running fix should clamp it to the most commonly used one", () => {
```

Comment:
grammar nit: "if the range is the range is outside" looks like duplication. Rephrase please.